pkg/kube/client.go

```diff
@@ -232,16 +233,11 @@ func (c *Client) Update(original, target ResourceList, force bool) (*Result, err
 			c.Log("Skipping delete of %q due to annotation [%s=%s]", info.Name, ResourcePolicyAnno, KeepPolicy)
 			continue
 		}
-
-		res.Deleted = append(res.Deleted, info)
 		if err := deleteResource(info); err != nil {
-			if apierrors.IsNotFound(err) {
```

Comment:
do we still want to check for this error type, or is the error message returned from `err` more than sufficient to handle this case?

I ask this because I do recall certain kubernetes API errors being quite cryptic in the past, which is why we were doing the error type check in the first place.